src/shell/Shell.cpp

```diff
@@ -540,6 +540,13 @@ static ValueRef* builtin262AgentSleep(ExecutionStateRef* state, ValueRef* thisVa
 
 class ShellPlatform : public PlatformRef {
 public:
+    bool canBlock;
+
+    ShellPlatform()
+        : canBlock(true)
```

Comment:
(Minor suggestion) How about encapsulating this `canBlock` member like defining it as private member and adding getter/setter methods?

Reply:
I can modify it

Reply:
I updated it